src/options/rule-delimiter.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+var gonzales = require('../gonzales');
+
+module.exports = (function() {
+    var valueFromSettings;
+    var blockIndent;
+    var value;
+
+    function findAtRules(node) {
+        var space = gonzales.createNode({type: 'space', content: value});
+
+        node.forEach('atruleb', function(atRuleNode, index) {
+            // For every atruleb - check for preceding space
+            var prevNode = node.get(index - 1);
+            if (prevNode) {
+                if (prevNode.is('space')) {
+                    prevNode.content = value;
+                    return;
+                } else {
+                    node.insert(index, space);
+                }
+            }
+        });
+    }
+
+    function findRuleSets(node) {
+        var space = gonzales.createNode({type: 'space', content: value});
+
+        node.forEach('ruleset', function(ruleSetNode, index) {
+            // For every ruleset - check for preceding space
+            var prevNode = node.get(index - 1);
+            if (prevNode) {
+                if (prevNode.is('space')) {
+                    prevNode.content = value;
```

Comment:
I see two cases here.
```scss
// Example (1)
.a {...} .b {...}

// Example (2)
.a {...}
   .b {...}
```

In case (1) you can just replace `prevNode.content` with `value` as you do now.
In case (2) you can replace all newlines in `prevNode.content` with `value`.
Something like: "let `y` be the index of the last newline, then replace content with `value` + substring from `y` to the end of space node". This should preserve indent without using `block-indent`.

Please let me know if I'm understandable at all :)